internal/compiler/passes/resolving.rs

```diff
@@ -607,6 +608,16 @@ impl Expression {
                     Expression::Invalid
                 }
             }
+            LookupResult::Expression { expression, .. }
+                if matches!(ctx.phase, LookupPhase::ResolvingTwoWayBindings)
+                    && matches!(expression, Expression::RepeaterModelReference { .. }) =>
```

Comment:
untested, but i thin you can simplify this like so

```suggestion
            LookupResult::Expression { expression: Expression::RepeaterModelReference { .. }, .. }
                if matches!(ctx.phase, LookupPhase::ResolvingTwoWayBindings)  =>
```

(And if using `#[derive(PartialEq)]` for the phase, you can use `==` instead of `matches!`)

Reply:
Excellent idea!